ci_framework/roles/ci_setup/tasks/cleanup.yml

```diff
@@ -5,3 +5,29 @@
   vars:
     directory_state: absent
   ansible.builtin.import_tasks: directories.yml
+
+- name: Disable the repos
+  become: true
+  tags:
+    - cleanup
+  when:
+    - cifmw_ci_setup_yum_repos is defined
+    - cifmw_ci_setup_yum_repos | length > 0
+  ansible.builtin.yum_repository:
+    name: "{{ item.name }}"
+    state: absent
+  loop: "{{ cifmw_ci_setup_yum_repos }}"
+
+- name: Disable the repos enabled in rhsm
+  become: true
+  when: ansible_distribution == 'RedHat'
+  vars:
+    _repos: >-
+      {{
+        cifmw_ci_setup_rhel_rhsm_default_repos +
+        ( cifmw_ci_setup_rhel_rhsm_extra_repos | default([]) )
+      }}
+  community.general.rhsm_repository:
```

Comment:
doesn't it intend the node is subscribed? Maybe we should check for subscription manager status before?